src/algebra/group/with_one.lean

```diff
@@ -29,6 +29,12 @@ def with_one (α) := option α
 
 namespace with_one
 
+instance [has_repr α] : has_repr (with_zero α) :=
+⟨λ o, match o with | none := "0" | (some a) := "(↑ " ++ repr a ++ ")" end⟩
```

Comment:
Wouldn't this be more accurate as
```suggestion
⟨λ o, match o with | none := "0" | (some a) := "↑(" ++ repr a ++ ")" end⟩
```
just in case `repr a` contains spaces

Reply:
(and throughout)

Reply:
No because all `repr`s have outer brackets. Have a look at `core.init.data.repr`. I'm happy to change this, but given that's in core I wouldn't do it so lightly.

Reply:
That's not true: consider the repr of `with_zero ℚ` on `1/2`. This prints it as `(↑ 1/2)`, but that's parsed by lean as `(↑ 1)/2 which isn't correct.

Reply:
Obviously this whole thing is fragile anyway, but if you're going to add brackets at all, they strike me as useless on the outside (since ↑ has such high binding power), but clearly there are cases where they make a difference on the inside.

Reply:
Sorry, what am I missing? I don't see `"("` in:

https://github.com/leanprover-community/mathlib/blob/2cb5edb5f13ec5b4ed5624d65a57768e710d709a/src/data/rat/basic.lean#L55-L57

Reply:
Ugh, I misread indeed. Then I would suggest putting outer parentheses in `rat.has_repr` to match all the `repr`s in core.

Reply:
They are all in one place: [`core.init.data.repr`](https://leanprover-community.github.io/mathlib_docs/init/data/repr.html)

Reply:
I'd still like to see examples of the outputs, not just the code that generates them. At a glance, `nat` and `bool` and `list` do not output parentheses, so I don't understand the argument you're making. Perhaps I would with examples.